.github/workflows/prod-deploy.yml

```diff
@@ -0,0 +1,20 @@
+name: Prod deploy
+
+on:
+  push:
+    tags:
+      - "v*"
+
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  deploy:
+    if: github.repository_owner == 'mdn' && github.ref_protected
```

Comment:
For the `github.ref_protected` part to work, I configured a tag ruleset that restricts updates, and deletions of all tags.

Note: As tags are currently created by github-actions, we cannot restrict creation without further ado, because we would need to add github-actions to the bypass list, and that does not appear to be possible.